app/app/users/[user_id]/page.tsx

```diff
@@ -0,0 +1,87 @@
+"use client";
+import Card from "react-bootstrap/Card";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Spinner from "react-bootstrap/Spinner";
+import Table from "react-bootstrap/Table";
+import AppBreadCrumb from "@/components/AppBreadCrumb";
+import { useToken } from "@/utils/auth";
+import { useUser } from "@/utils/users";
+import { User } from "@/types/users";
+import { formatDateTime } from "@/utils/formatters";
+
+type UserColumn = {
+  name: keyof User;
+  label: string;
+  renderer?: (user: User) => string;
+};
+
+// todo: this can be extended and used for the user list table column sorting, etc
+const COLUMNS: UserColumn[] = [
+  {
+    name: "app_metadata",
+    label: "Role",
+    renderer: (user) => user.app_metadata.roles[0] || "",
+  },
+  { name: "name", label: "Name" },
+  { name: "email", label: "Email" },
+  { name: "last_ip", label: "Last IP address" },
+  { name: "logins_count", label: "Login count" },
+  {
+    name: "created_at",
+    label: "Created at",
+    renderer: (user) => formatDateTime(user.created_at),
+  },
+  {
+    name: "last_login",
+    label: "Last login",
+    renderer: (user) => formatDateTime(user.last_login),
+  },
+  {
+    name: "updated_at",
+    label: "Updated at",
+    renderer: (user) => formatDateTime(user.updated_at),
+  },
+];
+
+export default function UserDetails({
```

Comment:
it looks like you are mostly using function declarations now across the app which i like, but there are a few components where you still use an arrow function like `CrashMap`. should we develop some kinda of standard for when to use function vs arrow or keep leaving that choice up to the dev

Reply:
yes, great point. i'd love to try and be consistent with this. i will run through the rest of the code and rework the declarations in a separate PR 🙏